hack/lib/util/docs.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+#
+# This library holds utility functions related to the generation
+# of manpages and docs.
+
+
+function generate_manual_pages() {
+  local dest="$1"
+  local cmdName="$2"
+  local filestore=".files_generated_${cmdName}"
+  local skipprefix="${3:-}"
+
+  # We do this in a tmpdir in case the dest has other non-autogenned files
+  # We don't want to include them in the list of gen'd files
+  local tmpdir="${OS_ROOT}/_tmp/gen_man"
+  mkdir -p "${tmpdir}"
+  # generate the new files
+  genman "${tmpdir}" "${cmdName}"
+  # create the list of generated files
+  ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/${filestore}"
+
+  # remove all old generated file from the destination
+  while read file; do
+    if [[ -e "${tmpdir}/${file}" && -n "${skipprefix}" ]]; then
+      local original generated
+      original=$(grep -v "^${skipprefix}" "${dest}/${file}") || :
+      generated=$(grep -v "^${skipprefix}" "${tmpdir}/${file}") || :
+      if [[ "${original}" == "${generated}" ]]; then
+        # overwrite generated with original.
+        mv "${dest}/${file}" "${tmpdir}/${file}"
+      fi
+    else
+      rm "${dest}/${file}" || true
+    fi
+  done <"${dest}/${filestore}"
+
+  # put the new generated file into the destination
+  find "${tmpdir}" -exec rsync -pt {} "${dest}" \; >/dev/null
+  #cleanup
+  rm -rf "${tmpdir}"
+
+  echo "Assets generated in ${dest}"
+}
+readonly -f generate_manual_pages
+
+function generate_documentation() {
```

Comment:
Maybe? These functions are not meant to be called anywhere else except `os::util::gen-docs` anyway.